src/components/ServerPaginationGrid.jsx

```diff
@@ -0,0 +1,77 @@
+import { useContext, useMemo, useState } from "react";
+import useSWR from "swr";
+import { DataGrid } from "@mui/x-data-grid";
+import LinearProgress from "@mui/material/LinearProgress";
+
+import { useMeasurements } from "../lib/hooks/swr-extensions";
+import { PreferenceContext } from "../pages/_app";
+
+const endpoint = "http://localhost:3000/api/v2/data?";
+const urlWithParams = (baseUrl, params) => baseUrl + new URLSearchParams(params);
```

Comment:
utility functions `urlWithParams`, `fetcher` may be moved to `lib` to be reusable